examples/nuxt-openai/nuxt.config.ts

```diff
@@ -2,7 +2,14 @@
 export default defineNuxtConfig({
   devtools: { enabled: true },
   modules: ['@nuxtjs/tailwindcss'],
+  nitro: {
+    preset: 'vercel-edge' // you can use 'vercel' or other providers here
+  },
   runtimeConfig: {
-    openaiApiKey: ''
+    openaiApiKey:
+      process.env.NUXT_OPENAI_API_KEY ||
+      (() => {
+        throw new Error('NUXT_OPENAI_API_KEY is not set!')
+      })()
```

Comment:
Agree! We could make sure which is effective for production build only but let's move validation topic to the nitro/nuxt core.

Reply:
(please keep `process.env.NUXT_OPENAI_API_KEY || ""` as replacement for best compatibility and readability)

Reply:
The reason I would advise we omit the env variable is that it can confuse people. ie. they may think that they can change it to `process.env.SOME_OTHER_KEY`. But of course that would only work at _build_ time when the config is read. By omitting it entirely it helps people understand the difference between the _default_ and the _runtime override_ value. Happy to discuss more later.

Reply:
Indeed, we could alternatively use a comment above the line to say how to override it, addressing readability concerns.

The second one is compatibility. Using platform environments is best but it is easy to mistakenly not provide the env when using local CLI to make deployment. Having process.env inside config makes sure it also works when env/token is provided at build-time.  This is also something i plan to consider for Nitro built-in behavior but worth to be included in this particular example because i think both me (ironically) and @Hebilicious initially made same mistake and it is a bad DX when something not works smooehtly.

Reply:
Let's instead auto-load variables in with nuxt/schema in 3.7 (separate from Nitro) so there is no difference between build/runtime. What do you think?

Reply:
I think this should go to nitro directly since it handles runtime env but that's also a good idea to also include possible validation. (i just don't want this PR to be blocked until then. either empty string or my suggestion above which is consistent to the proposed behavior are good to me :)